### DISCUSSION TITLE
Dịch unsupervised pretraining, greedy layer-wise pretraining, suboptimal

### DISCUSSION THREAD

**Discussion by quangnhat185** [mlbvn/handson-ml2-vn#111] [Terminology] [answered]
Suggestion của mình để dịch ba từ này như sau:

- `unsupervised pretraining`: `tiền huấn luyện không giám sát`
- `greedy layer-wise pretraining`: `tiền huấn luyện tham lam theo lớp`
- `suboptimal`: `cận tối ưu`

Context:
- unsupervised pretraining
> If you can gather plenty of unlabeled training data, you can try to use it to train an unsupervised model, such as an autoencoder or a generative adversarial network . Then you can reuse the lower layers of the autoencoder or the lower layers of the GAN’s discriminator, add the output layer for your task on top, and fine-tune the final network using supervised learning (i.e., with the labeled training examples)

- `greedy layer-wise pretraining`
>Note that in the early days of Deep Learning it was difficult to train deep models, so people would use a technique called greedy layer-wise pretraining . They would first train an unsupervised model with a single layer, typically an RBM, then they would freeze that layer and add another one on top of it, then train the model again (effectively just training the new layer), then freeze the new layer and add another layer on top of it, train the model again, and so on.

cc: @mlbvn/handson-ml

**Comment by ngcthuong** [comment]
Layer-wise là theo "từng lớp" mới đúng chứ nhỉ.

**Reply by quangnhat185** [reply]
Em cũng đang lăn tăn từ này, mà hiện tại trong glossary mình dang dịch `layer` là `tầng,` chắc nên sửa lại thành `từng tầng`

**Comment by minhduc0711** [comment]
Từ `suboptimal` mình nghĩ nên dịch là `không tối ưu` https://www.merriam-webster.com/dictionary/suboptimal

**Reply by cuongvng** [reply]
em thấy dịch như anh @quangnhat185 đúng nghĩa hơn.

**Reply by minhduc0711** [reply]
định nghĩa của từ này đâu phải "gần tối ưu" nhỉ
![image](https://user-images.githubusercontent.com/34294427/130316049-45d487f7-5e40-4f77-9903-357c81a37736.png)

**Reply by ngcthuong** [reply]
Cận tối ưu. Trong research dùng theo nghĩa này. Hoặc mình luôn hiểu theo nghĩa này 🥺

**Reply by cuongvng** [reply]
em lại hiểu nghĩa "less than optimal" thiên về "gần tối ưu"  hơn là "không tối ưu" 👀 👀 

`chưa tối ưu` thì sao nhỉ?

**Reply by minhduc0711** [reply]
Anh @tiepvupsu cho em xin ý kiến về từ này với ạ

**Reply by ngcthuong** [reply]
không phải là phương án tốt nhất nhưng cũng gần phương án tốt nhất. Còn dùng từ "không tối ưu" nó là theo ý nghĩa khác rồi.

**Reply by tiepvupsu** [reply]
Một vote cho "cận tối ưu" nhé. "không tối ưu" hay "chưa tối ưu nó mang nghĩ nagetive rồi.

**Comment by minhduc0711** [comment, marked as answer]
Vậy mình chốt như vậy nhé
- `unsupervised pretraining`: `tiền huấn luyện không giám sát`
- `greedy layer-wise pretraining`: `tiền huấn luyện tham lam theo từng tầng`
- `suboptimal`: `cận tối ưu`
